src/Support/ModuleRegistry.php

```diff
@@ -69,50 +63,34 @@ public function moduleForClass(string $fqcn): ?ModuleConfig
 	
 	public function modules(): Collection
 	{
-		if (null === $this->modules) {
-			$this->modules = $this->loadModules();
+		if (null === $this->modules && ($modules = $this->auto_discovery->modules())) {
+			$this->modules = $modules->mapWithKeys(function(array $module) {
+				return [
+					$module['name'] => new ModuleConfig($module['name'], $module['base_path'], new Collection($module['namespaces'])),
+				];
+			});
```

Comment:
Passing a closure in here would be nice, too, although this is `mapWithKeys` rather than each…